components/suggest/src/db.rs

```diff
@@ -132,31 +132,70 @@ impl<'a> SuggestDao<'a> {
             named_params! {
                 ":keyword": keyword,
             },
-            |row| -> Result<Suggestion> {
-                let keywords: Vec<String> = self.conn.query_rows_and_then(
+            |row| -> Result<Suggestion>{
+                let suggestion_id: i64 = row.get("id")?;
+                let title = row.get("title")?;
+                let url = row.get("url")?;
+                let icon = row.get("icon")?;
+                let provider = ProviderType::from_u8(row.get::<_, u8>("provider")?);
```

Comment:
```suggestion
                let provider = row.get("provider")?);
```

Here's what was causing your problem—a one-liner that makes all the difference! 😅 Instead of writing out the `ProviderType::from_u8` conversion yourself, you can write `row.get::<_, ProviderType>("provider")`, which will call your new `FromSql::column_result` conversion!

Bonus: you don't need the explicit `::<_, ProviderType>` annotation, either! If you change the match arms to just `ProviderType::Adm` and `ProviderType::Wikipedia`, Rust's type inferencer will figure out that `provider` must be a `ProviderType`.